internal/scheduler/internaltypes/resource_list_factory.go

```diff
@@ -64,15 +69,30 @@ func (factory *ResourceListFactory) FromNodeProto(resources schedulerobjects.Res
 	return ResourceList{resources: result, factory: factory}
 }
 
-// Fail on unknown resources, round up.
-func (factory *ResourceListFactory) FromJobProto(resources schedulerobjects.ResourceList) (ResourceList, error) {
+// Ignore unknown resources, round up.
+func (factory *ResourceListFactory) FromJobResourceListIgnoreUnknown(resources map[string]k8sResource.Quantity) ResourceList {
 	result := make([]int64, len(factory.indexToName))
-	for k, v := range resources.Resources {
+	for k, v := range resources {
 		index, ok := factory.nameToIndex[k]
+		if ok {
```

Comment:
do we want to log (even if only at debug) what we are ignoring here?

Reply:
We discussed this, decided against it as it would require passing ctx down, and we check at validation time anyway.